controllers/cluster_controller.go

```diff
@@ -262,9 +270,135 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 		}
 	}
 
+	if cluster.IsPackagesEnabled() {
+		if err := r.postReconcilePackagesForWorkloadCluster(ctx, log, cluster); err != nil {
+			return controller.Result{}, err
+		}
+	}
+
 	return controller.Result{}, nil
 }
 
+// PackagesClient handles curated packages operations from within the cluster
+// controller.
+type PackagesClient interface {
+	EnableFullLifecycle(context.Context, logr.Logger, string, string, *v1alpha1.Image, *registrymirror.RegistryMirror, ...curatedpackages.PackageControllerClientOpt) error
```

Comment:
Should we opt for defining the variables that will be consumed by this function? That will make the readibility easier.

Reply:
> Should we opt for defining the variables that will be consumed by this function? That will make the readibility easier.

Please see line 99 of this file for that information. I didn't want to have duplicates because then if I change the name in one place, I'll undoubtedly forget to change the name in the other, then I have an even bigger readability problem on my hands.

I generally only have names on the implementations, not the interfaces.